### ISSUE TITLE
"Not seekable" error when opening /dev/tty

### ISSUE THREAD

**Issue by stefanholek** [stefanholek/term#1]
Python 3.2.3
Linux 2.6.26 + 2.6.32

"Not seekable" error when opening /dev/tty:

```
    fd = os.open('/dev/tty', os.O_RDWR|os.O_NOCTTY)
    tty = os.fdopen(fd, 'w+', 1)
```

The second line raises an exception:

```
    Traceback (most recent call last):
    File "/home/stefan/sandbox/term/build/lib/term/tests/test_term.py", line 98, in test__opentty
        tty = _opentty('/dev/tty', 'w+', 1)
    File "/home/stefan/sandbox/term/build/lib/term/__init__.py", line 92, in _opentty
        return os.fdopen(fd, mode, bufsize)
    File "/home/stefan/sandbox/term/lib/python3.2/os.py", line 814, in fdopen
        return io.open(fd, *args, **kwargs)
    io.UnsupportedOperation: File or stream is not seekable.
```

Observations:
- This code works in Python 2 on Mac OS X and Linux.
- This code works on Mac OS X in all Python versions.
- On OS X I can also call os.lseek on the fd from line 1. On Linux the attempt raises OSError [Errno 29] Illegal seek.

**Comment by stefanholek** [comment]
In Python 3 the only way to open /dev/tty under Linux appears to be 1) in binary mode and 2) with buffering disabled.

```
fd = os.open('/dev/tty', os.O_RDWR|os.O_NOCTTY)
tty = open(fd, 'wb+', buffering=0)
```

**Comment by stefanholek** [comment]
This was fixed in 2.0.
